README.md

```diff
@@ -1,9 +1,259 @@
 [build-status]: https://travis-ci.org/kaazing/netx.ws
 
-WebSocket RFC-6455 implementation
+netx.ws is a Java-based implementation of [WebSocket RFC 6455](https://tools.ietf.org/html/rfc6455) client with following
+important characteristics:
+
+* Target JVM Version - Java SE 1.6
+* `java.net.Socket` and `java.net.URLConnection` style APIs
+* API for streaming binary and text payloads across WebSocket frames
+* API for sending and receiving text and binary messages
+* Extension SPI for extension developers
+* Authentication
+* HTTP Redirect Policies
+* Thread-safe
+* Garbage-free
+* Wait-free
+
+> Using the following maven dependency, developers can include netx.ws to develop and run web applications:
+
+```xml
+<dependency>
+    <groupId>org.kaazing</groupId>
+    <artifactId>netx.ws</artifactId>
+    <version>[0.1,1.0)</version>
+</dependency>
+```
+
+## Building this Project
+
+### Minimum requirements for building the project
+
+* Java SE Development Kit (JDK) 7 or higher
+* maven 3.0.5 or higher
+
+### Steps for building this project
+
+0. Clone the repo: ```git clone https://github.com/kaazing/netx.ws.git```
+0. Go to the cloned directory: ```cd netx.ws```
+0. Build the project: ```mvn clean install```
+
+## Target JVM Version - Java SE 1.6
+
+netx.ws is compiled using `-target=1.6` in order that the generated classes can be used to develop both desktop and
+mobile (Android) apps using the same jars.
+
+## java.net.Socket and java.net.URLConnection Style APIs
+
+netx.ws offers two options for creating and using WebSocket connections that enable developers to leverage their
+`java.net.Socket` and `java.net.URLConnection` experience:
+
+ * `org.kaazing.netx.ws.WebSocket` - for developers familiar with the `java.net.Socket` class and it's APIs.
+ * `org.kaazing.netx.ws.WsURLConnection` - for developers familiar with the `java.net.URLConnection` and it's APIs.
+
+### org.kaazing.netx.ws.WebSocket
+
+netx.ws offers `org.kaazing.netx.ws.WebSocketFactory` to create instances of `org.kaazing.netx.ws.WebSocket` as shown below:
+
+ ``` java
+ WebSocketFactory factory = WebSocketFactory.newInstance();
+ WebSocket connection = factory.createWebSocket(URI.toString("http://echo.websocket.org"));
+ ```
+### org.kaazing.netx.ws.WsURLConnection
+
+The following code shows how to create an instance of `org.kaazing.netx.ws.WsURLConnection` using
+`org.kaazing.netx.URLConnectionHelper`:
+
+``` java
+URLConnectionHelper helper = URLConnectionHelper.newInstance();
+WsURLConnection connection = (WsURLConnection) helper.openConnection(URI.create("ws://echo.websocket.org"));
+```
+
+## Stream Binary and Text Payloads Across WebSocket Frames
+
+Both `org.kaazing.netx.ws.WebSocket` and `org.kaazing.netx.ws.WsURLConnection` support APIs for streaming both binary and text
+payloads by filtering out
+
+* specific bits related to WebSocket framing such as opcode, mask, length, etc.
+* WebSocket Control Frames such as `CLOSE`, `PING`, or `PONG` in their entirety.
```

Comment:
Recommend you describe this slightly higher level, more about what data (information) is observed than what is omitted.  Likely and application developer will think in terms of WebSocket messages, whereas you, the library developer, will think more in terms of WebSocket frames. :smile: